### DISCUSSION TITLE
Option to treat newline as either soft or hard break

### DISCUSSION THREAD

**Discussion by borekb** [notaapp/talk#345] [Feature Requests]
Famously,

```md
First line<ENTER>
Second line
```

can be rendered as both:

- Soft break (most Markdown implementations): 
    ```
    First line Second line
    ```
- Hard break (GitHub's issues, pull requests, comments, discussions, ...):
    ```
    First line
    Second line
    ```

It's unfortunate that GitHub decided to do this but it's such a major Markdown-based platform that it cannot be ignored. I'd love if there was a way in Nota to have an option to treat newline as GitHub does – right now, it follows "only" CommonMark which always treats those as soft line breaks.

I'd even think about having a "pragma" in each file so that the behavior could be per-file. For example:

```md
<!-- newline=hard-break -->

This document would render as on GitHub,
i.e., with hard breaks.
```

The option should affect at least two things:

- Preview
- What is copied into clipboard when _Copy as Rich Text_ is used

What do you think? I know this is controversial but I've seen some MD editors do this (not per-file pragma but a global option, usually).

**Comment by astoilkov** [comment]
Yeah, it makes sense. I've converted the issue into a discussion so people can vote. On one side we don't want too many configuration options on the other this seems useful for some people. Also, GitHub rendering seems more intuitive (at least for me).

I will leave the discussion for some time to see if people will vote for this.
